grc/gui_qt/components/flowgraph_view.py

```diff
@@ -61,7 +62,13 @@ def createActions(self, actions):
         log.debug("Creating actions")
 
     def contextMenuEvent(self, event):
-        super(FlowgraphView, self).contextMenuEvent(event)
+        if self.scene().selected_blocks():
+            block = self.scene().selected_blocks()[0]
+            self.scene().clearSelection()
+            block.setSelected(True)
+            block.context_menu(event.globalPos())
+        else:
+            super(FlowgraphView, self).contextMenuEvent(event)
```

Comment:
I think it would be great if something happened even if no blocks are selected. That way, a new user will get the idea that the right-click button can be used for something. GRC Gtk opens (a subset of) the Edit menu, which is an option. I tried replacing the `super()` call with something like
```python
empty_menu = QtWidgets.QMenu()
no_action = QtWidgets.QAction("No block selected", enabled=False)
empty_menu.addAction(no_action)
empty_menu.exec_(event.globalPos())
```

I think it looked okay. Open to suggestions here

Reply:
Yes, this works for me. It looks good and acts like a special messages. But if we are right-clicking  in the canvas , we are on the fg level.  So why not open the file menu from the mainwindow as context menu? Here we get some usefull items.
In GRC GTK we get the edit menu with all items but 'more'  disabled and the 'more' submenu only contains disabled items.

So, does the File menu  bring any benefit to the user at this point? Otherwise I'd prefer your solution

Reply:
Interesting thought, but I think I'd prefer the empty menu. I don't think I (as a user) would think to right-click the canvas as a shortcut to the File menu

Reply:
I added the empty menu.